src/redis_connection.cc

```diff
@@ -426,6 +426,12 @@ void Connection::ExecuteCommands(const std::vector<Redis::CommandTokens> &to_pro
     svr_->SlowlogPushEntryIfNeeded(current_cmd_->Args(), duration);
     svr_->stats_.IncrLatency(static_cast<uint64_t>(duration), cmd_name);
     svr_->FeedMonitorConns(this, cmd_tokens);
+
+    // Break the execution loop when occurring the blocking command like BLPOP or BRPOP,
+    // it will suspend the connection and wait for the wakeup signal.
```

Comment:
Do wakeup signals here include new command sent by users?
I am unfamiliar with these code so correct me if I am wrong.

Reply:
No, it will remove the read callback: https://github.com/apache/incubator-kvrocks/blob/unstable/src/redis_cmd.cc#L1574, we can prevent the `current_cmd` being free in this case, even though it's a bit tricky.

Reply:
Got it. I understood the solution now.

Reply:
Cool, I think we should add some comments/docs for how the blocking pop works later to make it clear. 

The basic mechanism is below:

1. new connection will only enable `EV_READ`
2. remove the read callback when entering the blocking mode and add the write/timer callback 
3. wake up the connection by making the socket writable(command push was called) or timeout event was triggered, then re-enable and reset read callback again.